src/events/broker.py

```diff
@@ -76,7 +77,7 @@ def __init__(self, charm) -> None:
         )
 
         # Fast exit after workload instantiation, but before any event observer
-        if BROKER.value not in self.charm.config.roles:
+        if not any(role in self.charm.config.roles for role in [BROKER.value, CONTROLLER.value]):
```

Comment:
**suggestion:** I've noticed this awkward check in a couple of places now. Could it be useful for **another** flag for `runs_kafka`?

Reply:
I like to think about this using sets: we want the intersection of both sets to be not empty
```suggestion
        if not set(self.charm.config.roles.split(",")) & {BROKER.value, CONTROLLER.value}:
        # or using .intersect() if you want to be explicit
```
I guess the readability is a matter of personal taste, though